components/automate-cli/cmd/chef-automate/restart.go

```diff
@@ -49,6 +97,227 @@ func runRestartServices(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func init() {
-	RootCmd.AddCommand(restartServicesCmd)
+type restartFromBastion interface {
+	getAutomateHAInfraDetails() (*AutomateHAInfraDetails, error)
+	isManagedServicesOn() bool
+	executeRemoteExecutor(*NodeTypeAndCmd, SSHUtil, *cli.Writer) (map[string][]*CmdResult, error)
+	printRestartCmdOutput(map[string][]*CmdResult, string, *sync.WaitGroup, *sync.Mutex, *cli.Writer)
+}
+
+type restartFromBastionImpl struct{}
+
+func NewRestartFromBastionImpl() *restartFromBastionImpl {
+	return &restartFromBastionImpl{}
+}
+
+func (rs *restartFromBastionImpl) getAutomateHAInfraDetails() (*AutomateHAInfraDetails, error) {
+	return getAutomateHAInfraDetails()
+}
+
+func (st *restartFromBastionImpl) isManagedServicesOn() bool {
+	return isManagedServicesOn()
+}
+
+func (rs *restartFromBastionImpl) executeRemoteExecutor(nodemap *NodeTypeAndCmd, sshUtil SSHUtil, writer *cli.Writer) (map[string][]*CmdResult, error) {
+	remoteExecutor := NewRemoteCmdExecutor(nodemap, sshUtil, writer)
+	cmdResult, err := remoteExecutor.Execute()
+	return cmdResult, err
```

Comment:
Instead of creating a new instance you can store an instance of `remoteExecutor` in the restartFromBastionImpl struct while you are initialising it. 
This will reduce the initialisation overhead of remoteExecutor  when you are calling executeRemoteExecutor multiple times in your code

Reply:
You can use ExecuteWithNodeMap for you use case

Reply:
done